Feature/docs (#1011)
rhayes777/PyAutoFit: docs/overview/statistical_methods.rst

Graphical Models
----------------

For inference problems consisting of many datasets, the model composition is often very complex. Model parameters
can depend on multiple datasets, and the datasets themselves may be interdependent.

> Graphical models concisely describe these model and dataset dependencies and allow for
> them to be fitted simultaneously, with a concise API and scientific workflow that ensures scalability to big datasets.

Reviewer: ```suggestion
Graphical models concisely describe these model and dataset dependencies, allowing them to be fitted simultaneously. This is achieved through a concise API and scientific workflow that ensures scalability to large datasets.
```

A full description of using graphical models is given below:

https://github.com/Jammy2211/autofit_workspace/blob/release/notebooks/features/graphical_models.ipynb

Hierarchical Models
-------------------

> Hierarchical models are where multiple parameters in the model are assumed to be drawn from a common distributio.

Reviewer: ```suggestion
Hierarchical models are where multiple parameters in the model are assumed to be drawn from a common distribution.
```

> The parameters of this parent distribution are themselves inferred from the data, which for many problems enables
> more robust and informative model fitting.

Reviewer: ```suggestion
The parameters of this parent distribution are themselves inferred from the data, , enabling more robust and informative model fitting for many problems.
```

> Anfull description of using hierarchical models is given below:

Reviewer: ```suggestion
A full description of using hierarchical models is given below:

```

Is the model too complex or too simple?

Model comparison answers to these questions. It amounts to composing and fitting many different models to the data
and comparing how well they fit the data.

A full description of using model comparison is given below:

https://github.com/Jammy2211/autofit_workspace/blob/release/notebooks/features/model_comparison.ipynb

Interpolation
-------------

It is common to fit a model to many similar datasets, where it is anticipated that one or more model parameters vary
smoothly across the datasets.

For example, the datasets may be taken at different times, where the signal in the data and therefore model parameters
vary smoothly as a function of time.

It may be desirable to fit the datasets one-by-one and then interpolate the results in order
to determine the most likely model parameters at any point in time.

**PyAutoFit**'s interpolation feature allows for this, and a full description of its use is given below:

https://github.com/Jammy2211/autofit_workspace/blob/release/notebooks/features/interpolate.ipynb

Search Grid Search
------------------

> A classic method to perform model-fitting is a grid search, where the parameters of a model are divided on to a grid of

Reviewer: ```suggestion
A classic method to perform model-fitting is a grid search, where the parameters of a model are divided onto a grid of

```

Author: Thanks for all the suggestions! Implemented :)
